File: production/shared/src/main/scala/kpn/shared/node/NodeNetworkReference.scala
package kpn.shared.node

import kpn.shared.NetworkType

case class NodeNetworkReference(
  networkType: NetworkType,
  networkId: Long,
  networkName: String,
  nodeDefinedInRelation: Boolean,
  nodeRoleConnection: Boolean,
  nodeIntegrityCheck: Option[NodeNetworkIntegrityCheck],
  routes: Seq[NodeNetworkRouteReference]
)

File: spikes/ngrx/src/app/api/node-network-reference.ts
import {NodeNetworkIntegrityCheck} from './node-network-integrity-check';
import {NodeNetworkRouteReference} from './node-network-route-reference';
import {NetworkType} from './network-type';

export interface NodeNetworkReference {
  readonly networkType: NetworkType;
  readonly networkId: number;
  readonly networkName: string;
  readonly nodeDefinedInRelation: boolean;
  readonly nodeConnection: boolean;
  readonly nodeRoleConnection: boolean;
  readonly nodeIntegrityCheck: NodeNetworkIntegrityCheck;
  readonly facts: string[];
  readonly routes: NodeNetworkRouteReference[];
}
